File: app/src/main/java/com/oscar/notficacionlistener/IO/WebService/Model/NoticiaItem.java
package com.oscar.notficacionlistener.IO.WebService.Model;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

/**
 * Created by Oscar Emilio Pérez Mtz on 06/01/2019.
 * operez@na-at.com.mx
 */
public class NoticiaItem {

    @SerializedName("title")
    @Expose
    private String title;
    @SerializedName("link")
    @Expose
    private String link;
    @SerializedName("guid")
    @Expose
    private Object guid;
    @SerializedName("pubDate")
    @Expose
    private String pubDate;
    @SerializedName("description")
    @Expose
    private String description;
    @SerializedName("token")
    @Expose
    private String token;
    @SerializedName("urlImage")
    @Expose
    private String urlImage;

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getLink() {
        return link;
    }

    public void setLink(String link) {
        this.link = link;
    }

    public Object getGuid() {
        return guid;
    }

    public void setGuid(Object guid) {
        this.guid = guid;
    }

    public String getPubDate() {
        return pubDate;
    }

    public void setPubDate(String pubDate) {
        this.pubDate = pubDate;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getToken() {
        return token;
    }

    public void setToken(String token) {
        this.token = token;
    }


    public String getUrlImage() {
        return urlImage;
    }

    public void setUrlImage(String urlImage) {
        this.urlImage = urlImage;
    }
}

File: app/src/main/java/com/oscar/notficacionlistener/IO/WebService/Model/NoticiaItem.kt
package com.oscar.notficacionlistener.IO.WebService.Model

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
import java.text.SimpleDateFormat
import java.util.*

/**
 * Created by Oscar Emilio Pérez Mtz on 06/01/2019.
 * operez@na-at.com.mx
 */
class NoticiaItem {

    @SerializedName("title")
    @Expose
    var title: String? = null
    @SerializedName("link")
    @Expose
    var link: String? = null
    @SerializedName("guid")
    @Expose
    var guid: Any? = null
    @SerializedName("pubDate")
    @Expose
    var pubDate: String? = null
    @SerializedName("description")
    @Expose
    var description: String? = null
    @SerializedName("token")
    @Expose
    var token: String? = null
    @SerializedName("urlImage")
    @Expose
    var urlImage: String? = null

    fun getDateFromPubDate(): Date? {
        val simpleDateFormat = SimpleDateFormat("EEE, dd MMM yyyy HH:mm:ss zzz", Locale.US)
        return simpleDateFormat.parse(pubDate)
    }

    fun getReadeablePubdDate(): String? {
        return getDateFromPubDate()?.run {
            val simpleDateFormat = SimpleDateFormat(" dd MMM yyyy HH:mm:ss", Locale.US)
            simpleDateFormat.format(this)
        }
    }
}
